dev/TabView/TabViewItem.cpp

```diff
@@ -107,6 +109,25 @@ void TabViewItem::OnIsSelectedPropertyChanged(const winrt::DependencyObject& sen
     UpdateWidthModeVisualState();
 
     UpdateCloseButton();
+    UpdateForeground();
+}
+
+void TabViewItem::OnForegroundPropertyChanged(const winrt::DependencyObject&, const winrt::DependencyProperty&)
+{
+    UpdateForeground();
+}
+
+void TabViewItem::UpdateForeground()
+{
+    // We only need to set the foreground state when the TabViewItem is in rest state and not selected.
+    if (!IsSelected() && !m_isPointerOver)
+    {
+        // If Foreground is set, then change icon and header foreground to match.
+        winrt::VisualStateManager::GoToState(
+            *this,
+            ReadLocalValue(winrt::Control::ForegroundProperty()) == winrt::DependencyProperty::UnsetValue() ? L"ForegroundNotSet" : L"ForegroundSet",
```

Comment:
@jevansaks how do we feel about this these days?

Reply:
@jevansaks or @MikeHillberg  Wanting to clear some of these old PRs out. Is DP::UnsetValue acceptable to use? I know there was some contention on this in the past.

Reply:
Properties in general don't have an "is set" concept. Properties that are implemented with a DependencyProperty appear to have one, do to DP.UnsetValue. But this code example is only seeing if the local value is set or not, rather than checking of the property is being set by any means. For example, maybe the property is "set", but it's set by a Style.

For properties we want to have some kind of unset state we usually talk about sentinel values. For reference types that's usually null, for value types we frequently use Nullable to make it a reference type, and sometimes we use magic values like -1, min/max-infinity, or Nan. (Nullable is the better API choice.)

Is there some other condition we can use here?